src/site/content/en/blog/mediastreamtrack-insertable-media-processing/index.md

```diff
@@ -0,0 +1,257 @@
+`MediaStreamTrack` as a collection of [streams](/streams/) (as defined by the WHATWG
+[Streams API](https://streams.spec.whatwg.org/)). These streams can be manipulated to introduce new
+components.
+
+Granting developers access to the video (or audio) stream directly allows them to apply modifications
+directly to the stream. In contrast, realizing the same video manipulation task with traditional methods
+requires developers to use intermediaries such as `<canvas>` elements.
+(For details of this type of process, see, for example,
+[video + canvas = magic](https://html5doctor.com/video-canvas-magic/).)
+
+## Use cases
+
+Use cases for insertable streams for `MediaStreamTrack` include, but are not limited to:
+
+- Video conferencing gadgets like "funny hats" or virtual backgrounds.
+- Voice processing like software [vocoders](https://en.wikipedia.org/wiki/Vocoder).
+
+## Current status {: #status }
+
+<div class="w-table-wrapper">
+
+| Step                                     | Status                   |
+| ---------------------------------------- | ------------------------ |
+| 1. Create explainer                      | [Complete][explainer]    |
+| 2. Create initial draft of specification | [In Progress][spec]      |
+| 3. Gather feedback & iterate on design   | [In progress](#feedback) |
+| 4. **Origin trial**                      | **[In progress][ot]**    |
+| 5. Launch                                | Not started              |
+
+</div>
+
+## How to use insertable streams for `MediaStreamTrack`
+
+### Enabling via chrome://flags
+
+To experiment with insertable streams for `MediaStreamTrack` locally, without an origin trial token,
+enable the `#enable-experimental-web-platform-features` flag in `chrome://flags`.
+
+### Feature detection
+
+You can feature-detect insertable streams for `MediaStreamTrack` support as follows.
+
+```js
+if ('MediaStreamTrackProcessor' in window && 'MediaStreamTrackGenerator' in window) {
+  // Insertable streams for `MediaStreamTrack` is supported.
+}
+```
+
+### Core concepts
+
+Insertable streams for `MediaStreamTrack` builds on concepts previously proposed by
+[WebCodecs](https://web.dev/webcodecs/) and conceptually splits the `MediaStreamTrack` into two
+components:
+
+- The `MediaStreamTrackProcessor`, which consumes a `MediaStreamTrack` object's source and generates a stream
+  of media frames, specifically [`VideoFrame`](https://w3c.github.io/webcodecs/#videoframe-interface) or
+  [`AudioFrame`](https://w3c.github.io/webcodecs/#audioframe-interface)) objects. You can think of this as a
+  track sink that is capable of exposing the unencoded frames from the track as a `ReadableStream`
+  and that exposes a control channel for signals going in the opposite direction.
+- The `MediaStreamTrackGenerator`, which consumes a stream of media frames and exposes a `MediaStreamTrack`
+  interface, so that it can be used in any sink a `MediaStreamTrack` is currently attached to. It takes
```

Comment:
```suggestion
  interface, so that it can be used in any sink a `MediaStreamTrack` interface.
  The `MediaStreamTrackGenerator` can be provided to any sink, just like a track from `getUserMedia()`.
  It takes
```